tools/memory_analysis.py

```
@@ -20,10 +20,14 @@ def memory_analysis():
     except Exception as e:
         print(f"Error: {e}")
 
+def check_platform_compatibility():
+    supported = True
+    warnings = []
 
-def main():
-    memory_analysis()
-
+    try:
+        import psutil
```

Ruff (0.11.9)

`psutil` imported but unused

Remove unused import: `psutil`

(F401)